### PR TITLE
Feature/ searchbar for idle members names

### LINKED ISSUE #224: Add search functionality on Availability Panel
Search boxes are designed and available with **display property as none**  please remove/change the display property and no need to design it again

### PR DESCRIPTION
**Pull request by jjagannn**
Closes #224 
Change has been made to include a searchBar to filter by idle members names.

Web view - with new changes:
image

Web view - with new changes and searchBar toggled:
image

Mobile view:
image

Mobile view with searchBar toggled:
image

### CODE REVIEW

**Comment by Pavangbhat** [comment]
Hi @jjagannn, some checks are failing please fix that
And we also need to have search functionality on the tasks section as well

File: src/components/availability-panel/drag-drop-context/DroppableComponent.tsx [hunk 1]
```diff
@@ -18,13 +18,22 @@ const DroppableComponent: FC<droppableComponent> = ({
   unAssignedTasks,
   idleMembers,
   isTaskOnDrag,
+  searchTerm,
+  searchTermUnassignedTask,
```
Comment (Pavangbhat) [inline src/components/availability-panel/drag-drop-context/DroppableComponent.tsx:22]: Its better to use **searchTermForTask** or **searchTermTask** 
Please change all occurrence of **searchTermUnassignedTask**

File: src/components/availability-panel/drag-drop-context/DroppableComponent.tsx [hunk 2]
```diff
@@ -18,13 +18,22 @@ const DroppableComponent: FC<droppableComponent> = ({
...
   <div>
     {droppableId === 'tasks' ? (
       <Droppable droppableId="tasks" isCombineEnabled={!isTaskOnDrag}>
         {(provided) => (
           <div ref={provided.innerRef}>
-            {unAssignedTasks.map((task, index) => {
+            {unAssignedTasks.filter((val) => {
+              if (searchTermUnassignedTask === '') {
+                return true;
+              } if (val.title.toLowerCase().includes(searchTermUnassignedTask.toLowerCase())) {
```
Comment (Pavangbhat) [inline src/components/availability-panel/drag-drop-context/DroppableComponent.tsx:32]: What does **val** mean here ?
Let's make it more meaningful such as **task** or **taskItem**

File: src/components/availability-panel/drag-drop-context/DroppableComponent.tsx [hunk 3]
```diff
@@ -53,7 +62,14 @@ const DroppableComponent: FC<droppableComponent> = ({
       <Droppable droppableId="members" isCombineEnabled={isTaskOnDrag}>
         {(provided) => (
           <div ref={provided.innerRef}>
-            {idleMembers.map((member, index) => (
+            {idleMembers.filter((val) => {
+              if (searchTerm === '') {
+                return true;
+              } if (val.toLowerCase().includes(searchTerm.toLowerCase())) {
```
Comment (Pavangbhat) [inline src/components/availability-panel/drag-drop-context/DroppableComponent.tsx:68]: Same here 
Please change **val** to something meaningful

File: src/components/availability-panel/drag-drop-context/index.tsx [hunk 4]
```diff
@@ -122,14 +125,16 @@ const DragDropcontext: FC<dragDropProps> = ({
                 >
                   Search
                 </span>
-                {toogleSearch && <input />}
+                { toogleSearch && <input placeholder="Search by tasks" value={searchTermUnassignedTask} onChange={(e) => setSearchTermUnassignedTask(e.target.value)} type="text" id="unassignedTasksSearch" />}
```
Comment (Pavangbhat) [inline src/components/availability-panel/drag-drop-context/index.tsx:128]: What is the signification of **id** here? and on line **158**

File: src/components/availability-panel/drag-drop-context/index.tsx [hunk 5]
```diff
@@ -141,8 +146,16 @@ const DragDropcontext: FC<dragDropProps> = ({
...
+                <span
+                  onClick={() => {
+                    setToogleSearchName(!toogleSearchName);
+                  }}
+                  aria-hidden="true"
+                  className={classNames.searchText}
+                >
+                  Search
+                </span>
+                {toogleSearchName && <input placeholder="Search by name" value={searchTerm} onChange={(e) => setSearchTerm(e.target.value)} type="text" id="idleMemberSearch" />}
```
Comment (Pavangbhat) [inline src/components/availability-panel/drag-drop-context/index.tsx:158]: Since we are repeating **input** let's make it a separate component

**Review by Pavangbhat — CHANGES_REQUESTED** [review]
I have requested a few changes please have look

File: src/components/availability-panel/drag-drop-context/index.tsx (hunk 5 above)
Comment (jjagannn) [reply src/components/availability-panel/drag-drop-context/index.tsx]: Sure @Pavangbhat , I'll create a separate component and re-use.

**Comment by whyDontI** [comment]
Good work on the issue @jjagannn, just make sure the UI is same as shown in the [designs](https://www.figma.com/file/krYzdCT45ikzDDZ2tsUQaS/Tasks-Assignment?node-id=1%3A2).

File: src/components/availability-panel/drag-drop-context/index.tsx (hunk 5 above)
Comment (jjagannn) [reply src/components/availability-panel/drag-drop-context/index.tsx]: @Pavangbhat  
I've converted to a separate component and pushed the code now.

**Comment by jjagannn** [comment]
> Good work on the issue @jjagannn, just make sure the UI is same as shown in the [designs](https://www.figma.com/file/krYzdCT45ikzDDZ2tsUQaS/Tasks-Assignment?node-id=1%3A2).

Thanks Nikhil.
I'm working on adhering to the design currently.

**Review by whyDontI — CHANGES_REQUESTED** [review]
Please make the width of the search bar the same as the task card.
Please make sure you replicate the designs as it is.

Screenshot 2021-12-30 at 1 05 42 AM

**Review by whyDontI — CHANGES_REQUESTED** [review]
Good work 👍🏽 
Can we add these features too,
- Escape button should clear the text
- [`Trim`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim) the search text, which will solve this problem

**Comment by jjagannn** [comment]
> Good work 👍🏽 Can we add these features too,
> 
> * Escape button should clear the text
> * [`Trim`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim) the search text, which will solve this problem
> 
> 

Sure Nikhil, I'll work on this.

File: src/components/availability-panel/drag-drop-context/searchInputComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React, { FC } from 'react';
+import { searchInputComponent } from '@/interfaces/availabilityPanel.type';
+
+const SearchInputComponent: FC<searchInputComponent> = (
+  {
+    value, placeholder, type, onChangeMethod,
+  },
+) => (
+  <div>
```
Comment (Pavangbhat) [inline src/components/availability-panel/drag-drop-context/searchInputComponent.tsx:9]: Any reason to wrap input around the **div** tag?

**Review by Pavangbhat — CHANGES_REQUESTED** [review]
Requesting a few changes :-

- [ ] Please make the search box as same size as task/member card on small screen
- [ ] Whenever the search result is not found then show the message to the user that no task found or no idle members found

image

File: src/components/availability-panel/drag-drop-context/index.tsx [hunk 7]
```diff
@@ -28,16 +29,42 @@ const DragDropcontext: FC<dragDropProps> = ({
...
+    setSearchTermTask(e.target.value as string);
+  };
+
+  const handleChangeMember = (e:any) => {
+    setSearchTermMember(e.target.value as string);
+  };
+
+  const onkeypressedTask = (event:any) => {
+    var code = event.charCode || event.keyCode;
+    if (code == 27) {
```
Comment (Pavangbhat) [inline src/components/availability-panel/drag-drop-context/index.tsx:55]: can't we avoid if condition here? and write it in a single line

File: src/components/availability-panel/drag-drop-context/index.tsx [hunk 8]
```diff
@@ -28,16 +29,42 @@ const DragDropcontext: FC<dragDropProps> = ({
...
+      console.log(event);
+      setSearchTermTask('');
+    }
+  }
+
+  const onkeypressedMember = (event:any) => {
+    var code = event.charCode || event.keyCode;
```
Comment (Pavangbhat) [inline src/components/availability-panel/drag-drop-context/index.tsx:62]: Please avoid using var. we should use const where ever possible

File: src/components/availability-panel/drag-drop-context/index.tsx [hunk 9]
```diff
@@ -28,16 +29,42 @@ const DragDropcontext: FC<dragDropProps> = ({
...
+    if (code == 27) {
```
Comment (Pavangbhat) [inline src/components/availability-panel/drag-drop-context/index.tsx:63]: can't we avoid if condition here? and write it in a single line

**Review by Pavangbhat — CHANGES_REQUESTED** [review]
image

Provide feedback for the user if there is no match in search results

**Review by Pavangbhat — CHANGES_REQUESTED** [review]
Also, there is a merge conflict so please take a pull from upstream develop and rebase it
https://github.com/Real-Dev-Squad/website-welcome/blob/main/CONTRIBUTING.md

**Comment by ivinayakg** [comment]
hey, @whyDontI and @Pavangbhat,
so I have been assigned to fix this and I have been going through the code, but just for the common grounds.
Can you give me some pointers about what is left here to do other than merge conflict?
Like some changes, you required apart from the ones mentioned above or anything else.

**Comment by harshith-venkatesh** [comment]
@ankushdharkar , Since this seems to be repeated and stale PR, should this be closed?

**Pull request closed without merging**